app/finesse_data/__init__.py

```diff
@@ -0,0 +1,34 @@
+import logging
+
+import requests
+
+
+class FinesseDataFetchException(Exception):
+    """Custom exception for errors in fetching data from finesse-data."""
+
+
+class EmptyQueryError(Exception):
+    """Raised when the search query is empty."""
+
+
+def fetch_data(finesse_data_url, query):
+    if not query:
+        logging.error("Empty search query received")
+        raise EmptyQueryError("Search query cannot be empty")
+    try:
+        response = requests.get(finesse_data_url)
+        response.raise_for_status()
+        files = response.json()
+        query = query.replace("\r\n", "").replace("\n", "")
```

Comment:
while you're at it, why not use fuzzywuzzy?

https://builtin.com/data-science/fuzzy-matching-python

you don't have to normalize your data and you can return "close enough" matches

```
>>> fuzz.token_sort_ratio("This is my search query","THIS IS SEARCH QUERY")
93
>>> fuzz.token_sort_ratio("This is my search query","THIS IS NOT THE SEARCH QUERY")
82
>>> fuzz.token_sort_ratio("This is my search query","ENTIRELY SOMETHING ELSE")
39
```

Reply:
Done. Although, the normalization was because of the risk of log injection while logging "query", so it is still needed.

Reply:
well, that's good thinking! 

but wouldn't you want to filter out anything else than punctuation, alphanumeric characters, spaces then? any control character is potentially dangerous in logs.

https://www.theregister.com/2023/08/09/ansi_escape_sequence_risks/

```
>>> re.sub('[^\w\s\d\"#\$%&\'\(\)\*\+,-\./:;<=>\?@\[\\\]\^_`{\|}~]+', '', 'Example query search éço!! - ™')
'Example query search éço!! - ™'
```

also, shouldn't that filtering happen at the flask endpoint entry on any text fields?

Reply:
Done.